src/Xrefcheck/Verify.hs

```diff
@@ -423,7 +424,11 @@ checkExternalResource :: VerifyConfig -> Text -> IO (VerifyResult VerifyError)
 checkExternalResource VerifyConfig{..} link
   | skipCheck = return mempty
   | otherwise = fmap toVerifyRes $ runExceptT $ do
-      uri <- mkURI link & maybe (throwError ExternalResourceInvalidUri) pure
+      uri' <- URIBS.parseURI URIBS.laxURIParserOptions (encodeUtf8 link)
```

Comment:
Let's also add a test to make sure links with `[` and `]` are properly processed.